src/main/java/com/rsr5/gocd/riemann_notifier/GoNotificationPlugin.java

```diff
@@ -171,11 +222,52 @@ protected GoPluginApiResponse handleStageNotification(
                 .requestBody());
 
         try {
-            riemann.event().
-                    service(this.service(json)).
-                    state(this.state(json)).
-                    description(json.toString()).
-                    send().
+            List<Event> events = new ArrayList<>();
+            StageData notification = new StageData(json);
+
+            events.add(riemann.event().
+                       service(notification.getPipelineServiceName()).
+                       state(notification.state).
+                       description(json.toString()).
+                       build());
+
+            if (notification.state.equals("Failed") ||
+                notification.state.equals("Passed")) {
+
+                // output stage timing - add all the stage timings together
+                // for a given (pipeline, run-id) pair to get total pipeline
+                // execution time
+                events.add(riemann.event().
+                           service("gocd:stage-duration").
+                           time(notification.start.getEpochSecond()).
+                           state(notification.state).
+                           metric(notification.getDurationInSeconds()).
+                           attribute("pipeline", notification.pipeline).
+                           attribute("stage", notification.stage).
+                           attribute("run-id", notification.counter).
+                           build());
+
+                // jobs in a stage can run in parallel so this metric isn't
+                // necessarily useful for summation, but enables analysis of
+                // what's making a stage take a long time (and what isn't),
+                // for example.
+                if (notification.jobs != null) {
```

Comment:
From the constructor of `StageData`, it looks like `jobs` is guaranteed to be non-null.